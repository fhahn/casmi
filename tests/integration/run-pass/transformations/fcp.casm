// transform, cmdline "--fcp"

CASM fcp

// constant propergation

init initR

function af : -> Int
function bf : -> Int

rule initR =
{
	bf := BVadd_result(32, 1, 1)
	af := 2*3 + 5

    program(self) := undef
}
