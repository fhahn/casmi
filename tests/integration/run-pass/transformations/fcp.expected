initR=STATEMENTS(LET CONST(ident c0__,num 32),STATEMENTS(LET CONST(ident c1__,num 1),STATEMENTS(LET CONST(ident c2__,num 2),STATEMENTS(LET CONST(ident c3__,num 3),STATEMENTS(LET CONST(ident c4__,num 5),STATEMENTS(LET CONST(ident c5__,SELF),STATEMENTS(LET CONST(ident c6__,UNDEF),STATEMENTS(LET CONST(ident c7__,num 11),PARBLOCK(STATEMENTS(UPDATE(ident bf,ident c2__),STATEMENTS(UPDATE(ident af,ident c7__),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(ident c5__,null)),ident c6__),null))),null)))))))))
